packages/dds/merge-tree/src/client.ts

```diff
@@ -130,11 +130,26 @@ export class Client extends TypedEventEmitter<IClientEvents> {
 	private readonly clientNameToIds = new RedBlackTree<string, number>(compareStrings);
 	private readonly shortClientIdMap: string[] = [];
 
+	/**
+	 * @param specToSegment - Rehydrates a segment from its JSON representation
+	 * @param logger - Telemetry logger for diagnostics
+	 * @param options - Options for this client. See {@link IMergeTreeOptions} for details.
+	 * @param getMinInFlightRefSeq - Upon applying a message (see {@link Client.applyMsg}), client purges collab-window information which
+	 * is no longer necessary based on that message's minimum sequence number.
+	 * However, if the user of this client has in-flight messages which refer to positions in this Client,
+	 * they may wish to preserve additional merge information.
+	 * The effective minimum sequence number will be the minimum of the message's minimumSequenceNumber and the result of this function.
+	 * If this function returns undefined, the message's minimumSequenceNumber will be used.
+	 *
+	 * @privateRemarks
+	 * - Passing specToSegment would be unnecessary if Client were merged with SharedSegmentSequence
+	 * - AB#6866 tracks a more unified approach to collab window min seq handling.
+	 */
 	constructor(
-		// Passing this callback would be unnecessary if Client were merged with SharedSegmentSequence
 		public readonly specToSegment: (spec: IJSONSegment) => ISegment,
 		public readonly logger: ITelemetryLoggerExt,
 		options?: IMergeTreeOptions & PropertySet,
+		private readonly getMinInFlightRefSeq: () => number | undefined = () => undefined,
```

Comment:
kinda hate this. I wouldn't care as much if client were internal. Open to suggestions here, but I do think it would be nice to fix this bug (we actually see hits of 0x54e in production, which should be an easy follow-up fix for sequence once this change is in)

Reply:
Oh I'm surprised Client is marked `@alpha` -- I marked it `@internal` and deprecated it at one point. I'm not sure where it would be exposed such that it got marked `@alpha` again automatically.

I agree it's not great, though it does appear to solve the bug and I don't see any obvious ways that would be cleaner/simpler.